sql/plan/into.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql/expression"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// Into is a node to wrap the top-level node in a query plan so that any result will set user-defined or others
+// variables given
+type Into struct {
+	UnaryNode
+	IntoVars []sql.Expression
+}
+
+func NewInto(child sql.Node, variables []sql.Expression) *Into {
+	return &Into{
+		UnaryNode: UnaryNode{child},
+		IntoVars:  variables,
+	}
+}
+
+func (i *Into) String() string {
+	p := sql.NewTreePrinter()
+	var vars = make([]string, len(i.IntoVars))
+	for j, v := range i.IntoVars {
+		vars[j] = fmt.Sprintf(v.String())
+	}
+	_ = p.WriteNode("Into(%s)", strings.Join(vars, ", "))
+	_ = p.WriteChildren(i.Child.String())
+	return p.String()
+}
+
+func (i *Into) DebugString() string {
+	p := sql.NewTreePrinter()
+	var vars = make([]string, len(i.IntoVars))
+	for j, v := range i.IntoVars {
+		vars[j] = sql.DebugString(v)
+	}
+	_ = p.WriteNode("Into(%s)", strings.Join(vars, ", "))
+	_ = p.WriteChildren(sql.DebugString(i.Child))
+	return p.String()
+}
+
+func (i *Into) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	span, ctx := ctx.Span("plan.Into")
+	defer span.Finish()
+
+	rowIter, err := i.Child.RowIter(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+	rows, err := sql.RowIterToRows(ctx, nil, rowIter)
+	if err != nil {
+		return nil, err
+	}
+
+	rowNum := len(rows)
+	if rowNum > 1 {
+		return nil, sql.ErrMoreThanOneRow.New()
+	}
+	if rowNum == 0 {
+		// a warning with error code 1329 occurs (No data), and make not change to variables
+		return sql.RowsToRowIter(sql.Row{}), nil
+	}
+
+	var rowValues = make([]interface{}, len(rows[0]))
+
+	for j, val := range rows[0] {
+		rowValues[j] = val
+	}
+
+	for j, v := range i.IntoVars {
```

Comment:
What happens if len(rows[0]) is less than len(i.IntoVars)? 

In mysql, this is the behavior and error message:
```
mysql> select 1, 2, 3 into @my1, @my2;
ERROR 1222 (21000): The used SELECT statements have a different number of columns
```

I don't think we're handling this case, but let me know if I just missed a check in the code.

Reply:
oops, I missed that. Thank you!